### ISSUE TITLE
Generate Resource and Image azure-arm.image: ERROR: -> Forbidden : Client address is not authorized and caller is not a trusted service.

### ISSUE THREAD

**Issue by greadtm** [actions/runner-images#9389]
### Description

We generate windows 2022 images as part of the vmsscaleset create.
This has been working well up through last month. We attempted to run the process yesterday and at the GenerateResourceandImage step, a pkr key vault is created and the release fails with the following error:
2024-02-21T14:14:40.2381921Z ==> azure-arm.image: ERROR: -> Forbidden : Client address is not authorized and caller is not a trusted service.
2024-02-21T14:14:40.2383882Z ==> azure-arm.image: Client address: x.x.x.x from unknown subnet
2024-02-21T14:14:40.2385671Z ==> azure-arm.image: Caller: appid=***;oid=29e1fbed-ab6d-4741-8d3f-48590f5943c0;iss=https://sts.windows.net/***

where the IP address is our self-hosted agent. I replaced x.x.x.x with our real IP.

We have a ubuntu process that is working fine but that uses ssh over 172.x address.

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

runner image is based in git clone of todays date from main.

### Is it regression?

yes. We were able to build as of Janmuary 08, 2024.

### Expected behavior

build completes and image generated without error.

### Actual behavior

see above.

### Repro steps

execute GenerateandResourceImage.ps1 step on our self hosted agent VM.

**Comment by sergei-pyshnoi** [comment]
Hello @greadtm . Unfortunately its hard to determinate what exactly goes wrong in your case. As you can see [our nightly build](https://dev.azure.com/github/runner-images/_build/results?buildId=134837&view=results) fished without error. Possible it can be specific network or security restriction or etc in you subscription .

**Comment by Alexey-Ayupov** [comment]
Hello @greadtm. I could not reproduce the issue. Please find below the parameters I have used to generate an image:
```
       -SubscriptionId $SubscriptionId `
       -ResourceGroupName $ResourceGroupName -ImageGenerationRepositoryRoot "./" `
       -ImageType "Windows2022" -azurelocation $location `
       -AzureClientId $CLIENT_ID `
       -AzureClientSecret $clientSecret `
       -AzureTenantId $AZURE_TENANT
```
with the next GitHub workflow:

```

on:
    workflow_dispatch

permissions:
  id-token: write
  contents: read

jobs:
  repro:
    runs-on: ubuntu-latest
    environment: testing-images-ci
    steps:
      - uses: actions/checkout@v4
        with:
          repository: actions/runner-images

      - name: Azure login
        uses: azure/login@v1
        with:
          client-id: ${{ secrets.CLIENT_ID }}
          tenant-id: ${{ secrets.TENANT_ID }}
          subscription-id: ${{ secrets.SUBSCRIPTION_ID }}
          enable-AzPSSession: true
      - name: run imagegeneration
        shell: pwsh
        run: |
           Import-Module ./helpers/GenerateResourcesAndImage.ps1
           GenerateResourcesAndImage <parameters provided above>

```
It is impossible for us to reproduce this steps in azure devops, but since the images for both platforms are the same and  the `GenerateResourcesAndImage.ps1` was not changed recently, I assume the platform does not matter.

To investigate the problem further, could you please try new deployment and if it fails send us the Deployment logs from Azure side. You can find them under resource group `Deployments` tab. [How to...](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/deployment-history?tabs=azure-portal)

**Comment by greadtm** [comment]
I do not believe this is a code issue with the PS. 
I have a RG in Azure that has a defaulting VNET. I have relaxed the service endpoints so there is no keyvault firewall.
I created a new VNET/subnet to look like the one we were using.
I setup the packer.json to specify, the VNET name, VNET subnet name, and build image resource group.

When I run the release/GenerateResourceandImage.ps1 with the proper params like I have done in the past, I get the following error:

Getting the certificate's URL ...
2024-02-27T20:39:54.8581459Z ==> azure-arm.image: -> Key Vault Name : 'pkrkvztifzt2p81'
2024-02-27T20:39:54.8583086Z ==> azure-arm.image: -> Key Vault Secret Name : 'packerKeyVaultSecret'
2024-02-27T20:39:54.9360793Z ==> azure-arm.image: ERROR: -> Forbidden : Client address is not authorized and caller is not a trusted service.
2024-02-27T20:39:54.9361545Z ==> azure-arm.image: Client address: x.x.x.x from unknown subnet
2024-02-27T20:39:54.9364202Z ==> azure-arm.image: Caller: appid=***;oid=29e1fbed-ab6d-4741-8d3f-48590f5943c0;iss=https://sts.windows.net/***/
2024-02-27T20:39:54.9364737Z ==> azure-arm.image: Vault: pkrkvztifzt2p81;location=eastus2

where x.x.x.x=private IP address of our on prem(in Azure) build agent.

I am not sure why the script is sourcing the private IP address of the hosted build server vs a public IP. Note: this is happening for windows 2022. I can run for ubuntu image fine. Although the latter seems to use ssh to connect.
